File: frontend/get-routes.js
const fs = require('fs');
const path = require('path');

const appDir = path.join(__dirname, 'src', 'app');
const routes = [];

function walk(dir, currentRoute) {
  const entries = fs.readdirSync(dir, { withFileTypes: true });
  for (const entry of entries) {
    if (entry.isDirectory()) {
      if (entry.name.startsWith('[')) continue; // skip dynamic routes for bulk smoke test
      if (entry.name.startsWith('(')) {
        walk(path.join(dir, entry.name), currentRoute);
      } else {
        walk(path.join(dir, entry.name), currentRoute + '/' + entry.name);
      }
    } else if (entry.name === 'page.tsx') {
      routes.push(currentRoute === '' ? '/' : currentRoute);
    }
  }
}

walk(appDir, '');
console.log(JSON.stringify(routes, null, 2));

File: get_routes.py
import os
import json

routes = []
for root, dirs, files in os.walk("frontend/src/app"):
    for file in files:
        if file == "page.tsx":
            rel_path = os.path.relpath(root, "frontend/src/app")
            routes.append(rel_path.replace("\\", "/"))

print(json.dumps(routes, indent=2))
